homeassistant/components/homewizard/switch.py

```diff
@@ -85,12 +85,14 @@ def available(self) -> bool:
 
         This switch becomes unavailable when switch_lock is enabled.
         """
-        return super().available and not self.coordinator.api.state.switch_lock
+        return (
+            super().available and not self.coordinator.data[SERVICE_STATE].switch_lock  # type: ignore
+        )
 
     @property
     def is_on(self) -> bool:
         """Return true if switch is on."""
-        return bool(self.coordinator.api.state.power_on)
+        return bool(self.coordinator.data[SERVICE_STATE].power_on)  # type: ignore
```

Comment:
If I remove `# type: ignore`, I wil get the mypy error `error: Item "None" of "Optional[Any]" has no attribute "switch_lock"`. I have no idea how to fix this. Same issue in the lines below.

Reply:
Did you try casting it to `State` first (needs `homewizard_energy` import)? Then mypy would know the actual type and probably wouldn't complain.